script/tasks/fundNewWalletOnAllChains.ts

```diff
@@ -0,0 +1,239 @@
+import { defineCommand, runMain } from 'citty'
+import { createPublicClient, createWalletClient, http, parseAbi } from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+
+import { getAllNetworks, getViemChainForNetworkName } from '../../utils/network'
+import gasZipChainIds from '../resources/gasZipChainIds.json'
+import { BigNumber, BigNumberish } from 'ethers'
+import axios from 'axios'
+import { mainnet } from 'viem/chains'
+import { network } from 'hardhat'
+
+const GAS_ZIP_ROUTER_MAINNET = '0x9e22ebec84c7e4c4bd6d4ae7ff6f4d436d6d8390'
+const testnets = [
+  'bsc-testnet',
+  'lineatest',
+  'mumbai',
+  'sepolia',
+  'localanvil',
+  'virtualtestnet',
+]
+
+/// TYPES ///
+type HexString = `0x${string}`
+
+type GasZipChainIds = {
+  [key: string]: {
+    networkName: string
+    gasZipChainId: number
+  }
+}
+
+// this script is designed to be executed on mainnet (only)
+// it will get a list of all networks we support (minus testnets) and send an equal USD
+// amount worth of native tokens to each of these target networks using Gas.zip protocol
+
+// call this script
+// ts-node ./script/tasks/fundNewWalletOnAllChains.ts --privKeyFundingWallet "$PRIVATE_KEY" --receivingWallet "$PAUSER_WALLET" --doNotFundChains "[97,80001]" --fundAmountUSD "5"
+
+const main = defineCommand({
+  meta: {
+    name: 'fund-new-wallet-on-all-chains',
+    description:
+      'Funds a wallet with equal value of native gas on all supported chains',
+  },
+  args: {
+    privKeyFundingWallet: {
+      type: 'string',
+      description: 'Private key of the funding wallet',
+      required: true,
+    },
+    receivingWallet: {
+      type: 'string',
+      description: 'Address of the receiving wallet',
+      required: true,
+    },
+    doNotFundChains: {
+      type: 'string',
+      description: 'An array with chainIds that should not be funded',
+      required: true,
+    },
+    fundAmountUSD: {
+      type: 'string',
+      description: 'The amount of USD that should be sent to every chain',
+      required: true,
+    },
```

Comment:
Gas costs on different networks can be quite different, would it make sense to have a default value for each chain? e.g. 1$ to Polygon but 5$ to BSC ...

Reply:
I thought about this but neglected it since we have so many places where we already have to maintain code when adding a new network.
With 5 USD on every network at least one transaction can be executed for sure and it's still not too much money bound.

By the way....POL hat >12000 gas price today, transactions were costing real money. In such an environment, a wallet funded with varying amounts may would not be able to act. I would rather play it safe, that's why I decided to go for this implementation. 


Would you prefer if I hardcoded values for each network?

Reply:
ok